Source/JavaScriptCore/runtime/JSObjectInlines.h

```diff
@@ -868,13 +868,74 @@ bool JSObject::fastForEachPropertyWithSideEffectFreeFunctor(VM& vm, const Functo
 
     Structure* structure = this->structure();
 
-    if (!structure->canPerformFastPropertyEnumeration())
+    if (!structure->canPerformFastPropertyEnumerationCommon())
         return false;
 
     structure->forEachProperty(vm, functor);
     return true;
 }
 
+template<typename Functor>
+void JSObject::forEachIndexedProperty(JSGlobalObject* globalObject, const Functor& functor)
+{
+    ASSERT(structure()->canPerformFastPropertyEnumerationCommon());
+    if (!canHaveExistingOwnIndexedProperties())
+        return;
```

Comment:
Let's remove this. If we would like to early return here, this should be placed in the caller side of this function.